code/mob/living/critter/small_animal.dm

```diff
@@ -2957,6 +2965,56 @@ var/list/mob_bird_species = list("smallowl" = /mob/living/critter/small_animal/b
 				return 2
 		return ..()
 
+	mouse_drop(atom/over_object as mob|obj)
+		//if this bat is attacking/chasing someone, they won't stop just because you point at blood. Come on.
+		if (src.ai.target)
+			return ..()
+
+		if (ishuman(over_object) && usr == over_object)
+			var/mob/living/carbon/human/H = over_object
+			if (H && !H.restrained() && !H.stat && in_interact_range(src, H))
+				src.drink_targets += H
+				src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/bat/drink_blood, list(src, src.ai.default_task))
+				src.ai.interrupt()
+				src.set_loc(H.loc)
+				src.visible_message("[usr] offers up [his_or_her(usr)] arm to feed [src].")
+
+		//stand next to bat, and point towards some blood, the bat will try to drink it
+		else if (istype(over_object, /obj/item/reagent_containers/) && BOUNDS_DIST(usr, src) == 0)
+			src.drink_targets += over_object
+			src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/bat/drink_blood, list(src, src.ai.default_task))
+			src.ai.interrupt()
+			src.visible_message("[usr] gestures towards [over_object] to try to get [src] to drink from it.")
+		else
+			boutput(usr, "[src] looks a bit too preoccupied for you to direct it anywhere.")
+			return ..()
+
+	critter_ability_attack(mob/target)
+		var/datum/targetable/critter/drink_blood/B = src.abilityHolder.getAbility(/datum/targetable/critter/drink_blood)
+		if (B && !B.disabled && B.cooldowncheck())
+			B.handleCast(target)
+			return TRUE
+
+	// in here we'll search for reagent containers with blood
+	seek_target(var/range = 7)
+		. = list()
+
+		for (var/obj/fluid/F in view(range, src))
+			if (F.name == "blood")
+				. += F
+
+		for (var/obj/item/reagent_containers/S in view(range, src))
```

Comment:
S?!?!!!

Reply:
This was all copied-over from previous AI, I did not question its insanity.